packages/postgraphile-core/__tests__/integration/DistinctOnPlugin.js

```diff
@@ -0,0 +1,45 @@
+module.exports = builder => {
+  builder.hook(
+    "GraphQLObjectType:fields:field:args",
+    (args, build, context) => {
+      const {
+        graphql: { GraphQLString, GraphQLList },
+        pgSql: sql,
+      } = build;
+
+      const {
+        Self,
+        scope: { fieldName },
+        addArgDataGenerator,
+      } = context;
+
+      if (!(Self.name === "Query" && fieldName === "allToys")) {
+        return args;
+      }
+
+      addArgDataGenerator(({ distinct }) => {
+        return {
+          pgQuery: queryBuilder => {
+            distinct?.map(field => {
+              const id = sql.fragment`${queryBuilder.getTableAlias()}.${sql.identifier(
+                field
+              )}`;
```

Comment:
Just a warning in case people think copying things out of the tests is a good idea.

```suggestion
              // THIS IS AN EXAMPLE FOR THE TESTS. DO NOT USE THIS IN REAL PRODUCTION CODE!
              // Instead you should use an enum to indicate the allowed identifiers; otherwise
              // you risk interacting with the system columns and maybe worse.
              const id = sql.fragment`${queryBuilder.getTableAlias()}.${sql.identifier(
                field
              )}`;
```